src/libcore/default.rs

```diff
@@ -115,6 +134,35 @@ pub trait Default: Sized {
     fn default() -> Self;
 }
 
+/// A helper function that, when imported, allows one to reduce duplication, by
+/// saying just `default()` instead of `Default::default()`, when the compile
+/// can infer the value type.
+///
+/// For example:
+/// ```
+/// #![feature(default_free_fn)]
+///
+/// use std::default::default;
+///
+/// #[derive(Default)]
+/// struct SomeOptions {
+///     foo: i32,
+///     bar: f32,
+/// }
+///
+/// fn main() {
+///     let options = SomeOptions { foo: 42, ..default() };
```

Comment:
This isn't a compelling example to illustrate the reduction in repetition provided by `std::default::default` since it contains 50% more "default" than if they had written `Default::default()` without an import. To illustrate why someone would want to use `std::default::default` consider providing an example that involves calling default() more than once.

Reply:
Replaced with an example of a config type with two levels of nesting.